include/CppUTest/CppUTestConfig.h

```diff
@@ -211,13 +213,31 @@ typedef unsigned long long cpputest_ulonglong;
 /* Define some placeholders to disable the overloaded methods.
  * It's not required to have these match the size of the "real" type, but it's occasionally convenient.
  */
+
 #if defined(CPPUTEST_64BIT) && !defined(CPPUTEST_64BIT_32BIT_LONGS)
-typedef struct { char dummy[16]; } cpputest_longlong;
-typedef struct { char dummy[16]; } cpputest_ulonglong;
+#define CPPUTEST_SIZE_OF_FAKE_LONG_LONG_TYPE 16
 #else
-typedef struct { char dummy[8]; } cpputest_longlong;
-typedef struct { char dummy[8]; } cpputest_ulonglong;
+#define CPPUTEST_SIZE_OF_FAKE_LONG_LONG_TYPE 8
 #endif
+
+struct cpputest_longlong
+{
+#if defined(__cplusplus)
+  cpputest_longlong() {}
+  cpputest_longlong(int) {}
+#endif
+  char dummy[CPPUTEST_SIZE_OF_FAKE_LONG_LONG_TYPE];
+};
+
+struct cpputest_ulonglong
+{
+#if defined(__cplusplus)
+  cpputest_ulonglong() {}
+  cpputest_ulonglong(int) {}
+#endif
+  char dummy[CPPUTEST_SIZE_OF_FAKE_LONG_LONG_TYPE];
+};
+
```

Comment:
@uecasm I'm not sure I can follow about the not working in C code (or how it does work when it does). Could you suggest a C test that would / wouldn't work, depending on typedef / no typedef? Obviously, whatever is tested in C at present, does work. And generally, it is possible to use structs in C (e.g. as arguments) without typedef.